src/main.rs

```diff
@@ -542,36 +542,44 @@ async fn send(args: SendArgs) -> anyhow::Result<()> {
     if let Some(addr) = args.common.magic_ipv6_addr {
         builder = builder.bind_addr_v6(addr);
     }
-    let endpoint_fut = builder.bind();
+
     // use a flat store - todo: use a partial in mem store instead
     let suffix = rand::thread_rng().gen::<[u8; 16]>();
-    let iroh_data_dir =
-        std::env::current_dir()?.join(format!(".sendme-send-{}", hex::encode(suffix)));
-    if iroh_data_dir.exists() {
-        println!("can not share twice from the same directory");
+    let cwd = std::env::current_dir()?;
+    let blobs_data_dir = cwd.join(format!(".sendme-send-{}", hex::encode(suffix)));
+    if blobs_data_dir.exists() {
+        println!(
+            "can not share twice from the same directory: {}",
+            cwd.display(),
+        );
         std::process::exit(1);
     }
-    let iroh_data_dir_2 = iroh_data_dir.clone();
-    let _control_c = tokio::spawn(async move {
-        tokio::signal::ctrl_c().await?;
-        std::fs::remove_dir_all(iroh_data_dir_2)?;
-        std::process::exit(1);
-        #[allow(unreachable_code)]
-        anyhow::Ok(())
-    });
-    std::fs::create_dir_all(&iroh_data_dir)?;
-    let db = iroh_blobs::store::fs::Store::load(&iroh_data_dir).await?;
+
+    tokio::fs::create_dir_all(&blobs_data_dir).await?;
+
+    let endpoint = builder.bind().await?;
+    let lp = LocalPool::default();
+    let ps = SendStatus::new();
+    let blobs = Blobs::persistent(&blobs_data_dir)
+        .await?
+        .events(ps.new_client().into())
+        .build(lp.handle(), &endpoint);
+
+    let router = iroh::protocol::Router::builder(endpoint)
+        .accept(iroh_blobs::ALPN, blobs.clone())
+        .spawn()
+        .await?;
+
     let path = args.path;
-    let (temp_tag, size, collection) = import(path.clone(), db.clone()).await?;
+    let (temp_tag, size, collection) = import(path.clone(), blobs.store().clone()).await?;
     let hash = *temp_tag.hash();
-    // wait for the endpoint to be ready
-    let endpoint = endpoint_fut.await?;
+
     // wait for the endpoint to figure out its address before making a ticket
-    while endpoint.home_relay().is_none() {
+    while router.endpoint().home_relay().is_none() {
```

Comment:
Can we replace this by using [watch_home_relay](https://docs.rs/iroh/latest/iroh/endpoint/struct.Endpoint.html#method.watch_home_relay)?

This whole sleep thing is not very nice.

Reply:
fixed